Core/include/Acts/Utilities/Result.hpp

```diff
@@ -28,7 +28,7 @@ class Result {
    * This is used by the factory static methods to set up
    * the variant unambiguously in all cases.
    */
-  Result(std::variant<T, E>&& var) : m_var(var) {}
+  Result(std::variant<T, E>&& var) : m_var(std::move(var)) {}
```

Comment:
Is this not dangerous? If you pass by ref, this silently kills the variable.

Reply:
It never passes by `const ref&` here since the argument is a true rvalue reference.

Reply:
Hm, could be that `&&` is a true rvalue reference in this case, then it might be safe.

Reply:
Ok fine.